### ISSUE TITLE
[BUG] LDAP Keys are not updating user

### ISSUE THREAD

**Issue by Stiles96** [Leantime/leantime#2000]
*If you have problems with the installation please use our [community forum](https://community.leantime.io)*

**What set up are you using**
* [ ] Cloud Hosted Version
* [x] Self Hosted

**Describe the bug**
After Login the LDAP Keys phone, jobTitle, jobLevel, departement will be cleared.

Config:
public $ldapKeys = '{
        "username":"sAMAccountName",
        "groups":"memberof",
        "email":"mail",
        "firstname":"givenname",
        "lastname":"sn",
        "phone":"telephoneNumber", //Changed from telephoneNumber to phone
        "jobTitle":"title",
        "jobLevel":"",
        "department":"department"
    }';

Error Log:
[18-Oct-2023 10:23:26 Europe/Berlin] PHP Warning:  Undefined property: stdClass::$phone in /var/www/leantime/app/Domain/Ldap/Services/Ldap.php on line 209
[18-Oct-2023 10:23:26 Europe/Berlin] PHP Warning:  Undefined property: stdClass::$phone in /var/www/leantime/app/Domain/Ldap/Services/Ldap.php on line 238

**To Reproduce**
1. Update Users Info
2. Sign out and in
3. Check Info
4. Infos are cleared and not overwritten from AD Attributes

**Expected behavior**
The User Info will be overwritten from LDAP / AD Attributes. The values in the User Object are correct. It was working until 2.3.X

**Leantime Version**
2.4.3

**Server**
Apache2

**PHP / MySQL Version**
PHP 8.1

**Comment by marcelfolaron** [comment]
Just to confirm, you are saying that the JobTitle, level and department were set in Leantime (assuming through the Leantime UI) and then overwritten by the ldap sync?

**Comment by marcelfolaron** [comment]
On the phone/phonenumber error, please make sure that the key for the phonenumber is actually called `phone` it used to be phonenumber but that was changed.

**Comment by Stiles96** [comment]
It will overwritten by null not by the ldap values.
I changed the ldapkey already. No change.

**Comment by marcelfolaron** [comment]
Can you turn on $debug and try again and check the output of the error.log ? When the user is pulled from the ldap directory we log the data to see what was returned (when debug is on). You should see something like:

```
error_log("LEANTIME: Testing the logging\n");

            error_log("LEANTIME: >>>Attributes Begin>>>>>>\n");
            error_log("LEANTIME: fn $firstname", 0);
            error_log("LEANTIME: sn $lastname", 0);
            error_log("LEANTIME: phone $phone", 0);
            error_log("LEANTIME: role $role", 0);
            error_log("LEANTIME: username $uname ", 0);
            error_log("LEANTIME: jobTitle $jobTitle ", 0);
            error_log("LEANTIME: jobLevel $jobLevel ", 0);
            error_log("LEANTIME: department $department ", 0);
            error_log("LEANTIME: >>>Attributes End>>>>>>\n", 0);
```

**Comment by Stiles96** [comment]
18-Oct-2023 15:14:35 Europe/Berlin] LEANTIME: >>>Attributes Begin>>>>>>

[18-Oct-2023 15:14:35 Europe/Berlin] LEANTIME: fn Pa****
[18-Oct-2023 15:14:35 Europe/Berlin] LEANTIME: sn Hen****
[18-Oct-2023 15:14:35 Europe/Berlin] LEANTIME: phone
[18-Oct-2023 15:14:35 Europe/Berlin] LEANTIME: role 50
[18-Oct-2023 15:14:35 Europe/Berlin] LEANTIME: username P****n@h****u
[18-Oct-2023 15:14:35 Europe/Berlin] LEANTIME: jobTitle
[18-Oct-2023 15:14:35 Europe/Berlin] LEANTIME: jobLevel
[18-Oct-2023 15:14:35 Europe/Berlin] LEANTIME: department
[18-Oct-2023 15:14:35 Europe/Berlin] LEANTIME: >>>Attributes End>>>>>>

**Comment by marcelfolaron** [comment]
Thanks. So it appears that it can't find those fields in your ldap directory. Are you sure title & department are the keys within your ldap directory?

**Comment by Stiles96** [comment]
![grafik](https://github.com/Leantime/leantime/assets/51234422/1d79f376-0ce1-47a2-9db2-5e76ddc22a46)
![grafik](https://github.com/Leantime/leantime/assets/51234422/2e872736-5557-4c78-8a19-e3ad407ee66c)

Of course. I shared the 2 attributes of my user profile in the AD.
It works already until the update to 2.4.X

**Comment by marcelfolaron** [comment]
Fixed in 2.4.4
